### PR TITLE
fix: restoring cfg.json #2847

### PR DESCRIPTION
**Pull request by ahadcove**
This is in response to https://github.com/Aircoookie/WLED/issues/2847

This issue will resolve not being able restore cfg.json using Backup & restore feature within the Security Settings page

### CODE REVIEW

**Comment by blazoncek** [comment]
String::operator== should be overloaded. Not sure how exactly. 😁

**Comment by ahadcove** [comment]
I switched to index of based on blazoncek recommendation.
Any issue that you all see with this?

**Comment by softhack007** [comment]
I think a simple test would tell us: Just download cfg and presets, then do a factory reset and restore presets (first), then restore cfg and check if everything is back as expected. OK, only Wifi credentials will be lost because they are in another hidden file.

**Comment by Aircoookie** [comment]
Thank you for the fix, changed it to equals for presets.json since `finalname` will always have the leading slash and it uses less memory this way (not both `/presets.json` and `presets.json` in RAM). And I don't want to over-use F() strings either in file/callback operations.

**Comment by blazoncek** [comment]
@Aircoookie oops!

File: wled00/wled_server.cpp [hunk 1]
```diff
@@ -39,20 +39,20 @@ void handleUpload(AsyncWebServerRequest *request, const String& filename, size_t
...
-      finalname = "/" + finalname; // prepend slash if missing
+      finalname = '/' + finalname; // prepend slash if missing
     }
 
     request->_tempFile = WLED_FS.open(finalname, "w");
-    DEBUG_PRINT("Uploading ");
+    DEBUG_PRINT(F("Uploading "));
     DEBUG_PRINTLN(finalname);
-    if (finalname.equals("/presets.json") >= 0) presetsModifiedTime = toki.second();
+    if (finalname.equals(F("/presets.json")) >= 0) presetsModifiedTime = toki.second();
```
Comment (Aircoookie) [inline wled00/wled_server.cpp:48]: Needs no `F()` as also present as ram string in presets.cpp

File: wled00/wled_server.cpp [hunk 2]
```diff
@@ -39,20 +39,20 @@ void handleUpload(AsyncWebServerRequest *request, const String& filename, size_t
...
-      finalname = "/" + finalname; // prepend slash if missing
+      finalname = '/' + finalname; // prepend slash if missing
     }
 
     request->_tempFile = WLED_FS.open(finalname, "w");
-    DEBUG_PRINT("Uploading ");
+    DEBUG_PRINT(F("Uploading "));
     DEBUG_PRINTLN(finalname);
-    if (finalname == "/presets.json") presetsModifiedTime = toki.second();
+    if (finalname.equals(F("/presets.json")) >= 0) presetsModifiedTime = toki.second();
```
Comment (softhack007) [inline wled00/wled_server.cpp:48]: I think that string.equals() returns bool, so should not be tested for ">=0"

**Comment by blazoncek** [comment]
🔥 
😁

File: wled00/wled_server.cpp (hunk 2 above)
Comment (Aircoookie) [reply wled00/wled_server.cpp:48]: 100% correct 😅

**Review by Aircoookie — APPROVED** [review]

**Comment by ahadcove** [comment]
> Thank you for the fix, changed it to equals for presets.json since `finalname` will always have the leading slash and it uses less memory this way (not both `/presets.json` and `presets.json` in RAM). And I don't want to over-use F() strings either in file/callback operations.

Great. Thanks for fixing!

**Review by blazoncek — APPROVED** [review]
Looks fine to me too.

**Pull request merged**